From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rubens Brandao <git@rubens.io>
Date: Wed, 7 Jun 2023 16:18:00 -0300
Subject: [PATCH] 5428: Fix AARCH64 fcadd and fcmla export len

Fix AARCH64 fcadd_rotate export len

Fix AARCH64 fcmla_rotate export len
---
 .../AARCH64/data/languages/AARCH64neon.sinc   | 20 +++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/Ghidra/Processors/AARCH64/data/languages/AARCH64neon.sinc b/Ghidra/Processors/AARCH64/data/languages/AARCH64neon.sinc
index 58f6137d7..66633d40a 100644
--- a/Ghidra/Processors/AARCH64/data/languages/AARCH64neon.sinc
+++ b/Ghidra/Processors/AARCH64/data/languages/AARCH64neon.sinc
@@ -3554,8 +3554,8 @@ is b_31=0 & b_30=1 & b_2129=0b101110010 & b_1015=0b000101 & Rd_VPR128.8H & Rn_VP
 
 # C7.2.46 FCADD page C7-1090 line 63037 KEEPWITH
 
-fcadd_rotate: #90 is b_12=0 { export 90:1; }
-fcadd_rotate: #270 is b_12=1 { export 270:1; }
+fcadd_rotate: #90 is b_12=0 { export 90:2; }
+fcadd_rotate: #270 is b_12=1 { export 270:2; }
 
 # C7.2.53 FCADD page C7-2127 line 123874 MATCH x2e00e400/mask=xbf20ec00
 # CONSTRUCT x2e40e400/mask=xffe0ec00 MATCHED 1 DOCUMENTED OPCODES
@@ -4270,14 +4270,14 @@ is b_31=0 & b_30=1 & b_1029=0b00111011111000110010 & Rd_VPR128.8H & Rn_VPR128.8H
 
 # C7.2.55 FCMLA (by element) page C7-1117 line 64749 KEEPWITH
 
-fcmla_rotate: #0 is b_15=0 & b_1314=0b00 { export 0:1; }
-fcmla_rotate: #90 is b_15=0 & b_1314=0b01 { export 90:1; }
-fcmla_rotate: #180 is b_15=0 & b_1314=0b10 { export 180:1; }
-fcmla_rotate: #270 is b_15=0 & b_1314=0b11 { export 270:1; }
-fcmla_rotate: #0 is b_15=1 & b_1112=0b00 { export 0:1; }
-fcmla_rotate: #90 is b_15=1 & b_1112=0b01 { export 90:1; }
-fcmla_rotate: #180 is b_15=1 & b_1112=0b10 { export 180:1; }
-fcmla_rotate: #270 is b_15=1 & b_1112=0b11 { export 270:1; }
+fcmla_rotate: #0 is b_15=0 & b_1314=0b00 { export 0:2; }
+fcmla_rotate: #90 is b_15=0 & b_1314=0b01 { export 90:2; }
+fcmla_rotate: #180 is b_15=0 & b_1314=0b10 { export 180:2; }
+fcmla_rotate: #270 is b_15=0 & b_1314=0b11 { export 270:2; }
+fcmla_rotate: #0 is b_15=1 & b_1112=0b00 { export 0:2; }
+fcmla_rotate: #90 is b_15=1 & b_1112=0b01 { export 90:2; }
+fcmla_rotate: #180 is b_15=1 & b_1112=0b10 { export 180:2; }
+fcmla_rotate: #270 is b_15=1 & b_1112=0b11 { export 270:2; }
 
 # C7.2.62 FCMLA (by element) page C7-2154 line 125620 MATCH x2f001000/mask=xbf009400
 # CONSTRUCT x2f401000/mask=xffc09c00 MATCHED 1 DOCUMENTED OPCODES
-- 
2.40.1

